MetaMask/metamask-mobile
# feat: Snaps new architecture mobile integration

## **Description**

This PR adds a refactored Snaps Webview approach and embedded execution environment to allow mobile app consume the new mobile snaps architecture.

## **Related issues**

Fixes: N/A 

## **After Screenshot**
https://github.com/MetaMask/metamask-mobile/assets/44679989/6e2308dd-4ad6-4f62-a875-fdd13b0c8528


## **Manual testing steps**

1. Open the mobile app
2. Navigate to explore session and access snaps example page
3. add any snap
4. go to snaps item on settings
5. check if the installed snap is listed.

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**


## **CI Related Section: DO NOT EDIT**

Flag used by Bitrise to either run or skip E2E smoke tests

- Bitrise Flag:[skip ci]
